### DISCUSSION TITLE
List all .tiff /.jpeg files with clip-mask available

### DISCUSSION THREAD

**Discussion by ciochu** [ImageMagick/ImageMagick#6208] [Help] [answered]
### ImageMagick version

7.1.1-5

### Operating system, version and so on

Windows

### Description

I'm looking for so solution to check multiple files in bulk for clipping path existence. 
For now the only option i have is to run transformation for all my tiff and jpegs and see if output files (png) have transparent bg. 

Is there any method to check if clip-mask exist and then return it in log.txt file or embed into EXIF metadata? 
I noticed that not all my files (generated by many media agencies) have ClippingPathName in EXIF. I wish to fix that.

**Comment by snibgo** [comment]
How about:
```
magick in.tif -verbose info: |grep "Clipping path"
```
If your files are small and few in number, use "*" as the input filename. Otherwise, put it in a shell loop.

**Reply by ciochu** [reply]
Thanks for helping. 
Just to clarify for other users in Windows i had to replace grep by findstr 
But the result is still not satisfying.... 

I just get Clipping path: and nothing else, but i'm sure that cp is there. 

`magick identify -verbose my_file.tiff | findstr "Clipping Path"
  Clipping path:
`

**Reply by ciochu** [reply]
chat GPT helped me a lot, but there might be area where it won't be able to solve mystery.

I'm quite lost why with that code
`mogrify -path converted -format png -clip -verbose *.tiff`
I can convert some tiffs to png with transparent BG, 

but my another code can not identify clipping paths which any combiation clip-mask, clip-path, clipping-path. 
Any idea how to modify that code to works?
`for /r "C:\Users\a7xtdzz\Downloads\export" %f in (*.tif *.png) do magick identify -quiet -format "%%[clip-mask:exists]\n" "%f" | findstr "True" >nul && echo "Plik %f ma clipping path." >> log2.txt`

**Comment by snibgo** [comment]
Your long command "`for /r`" has one `%` doubled (as needed for BAT scripts) but others not (as needed for CMD commands). In addition, `clip-mask:exists` is not a property usually known to IM.

Did you get that from GPT? My advice is: don't. I won't waste my time debugging code from GPT or any other AI.

**Reply by ciochu** [reply]
@snibgo I fully understand, but i won't be sorry for using GPT to look for potential solution is there is none available in any documentation :)
I just wanted to check which attribute can return 0/1 checking if clipping path exist ;) Thank so much you for answering my question in another ticket. 

https://github.com/ImageMagick/ImageMagick/discussions/6212

**Comment by snibgo** [comment, marked as answer]
> I just wanted to check which attribute can return 0/1 checking if clipping path exist 

There is no attribute that returns 0 or 1 for whether a clipping path exists.

As I showed in your other thread https://github.com/ImageMagick/ImageMagick/discussions/6208 you can search the stdout output from verbose info for the string "`Clipping path:`".

The text that follows "`Clipping path:`" is an SVG version of the clipping path. You can get this directly if you want, so it gives the text of the clipping path, or no text if there is no clipping path.
```
magick identify -format %[8BIM:1999,2998:#1] toes.png

identify: unknown image property "%[8BIM:1999,2998:#1]" @ warning/property.c/InterpretImageProperties/4229.
```
(The warning is sent to stderr, not stdout.)

`toes.png` has no clipping path, so nothing is sent to stdout. But for your file, we get this:
```
magick identify -format %[8BIM:1999,2998:#1] 1072828_-_VHB_5962.tiff[0]

<?xml version="1.0" encoding="iso-8859-1"?>
<svg xmlns="http://www.w3.org/2000/svg" width="2700" height="2700">
<g>
<path fill-rule="evenodd" style="fill:#000000;stroke:#000000;stroke-width:0;stroke-antialiasing:false" d="
M 1336.74 2644.51
L 1332.52 2653.03
L 1338.15 2657.89
L 1333.93 2662.76
C 1333.93 2662.76 1404.45 2662.38 1438.2 2655.46
C 1471.94 2648.54 1646.5 2618.69 1656.6 2618.97
C 1656.6 2618.97 1755.69 2599.81 1841.19 2556.95
C 1926.7 2514.09 2110.54 2366.85 2203.33 2203.04
C 2296.11 2039.23 2362.19 1895.27 2399.19 1586.43
C 2399.19 1586.43 2415.49 1397.84 2402.01 1306.71
C 2402.01 1306.71 2388.49 1114.62 2369.6 1065.9
C 2369.6 1065.9 2338.53 921.233 2321.69 872.53
C 2304.85 823.827 2257.46 692.594 2146.96 516.187
C 2036.47 339.779 1872.84 184.749 1663.65 100.251
C 1663.65 100.251 1524.53 41.0549 1373.38 51.6035
C 1373.38 51.6035 1299.37 51.785 1208.52 50.3873
C 1117.67 48.9895 887.147 53.9597 681.522 268.085
C 475.898 482.21 418.455 695.086 384.207 797.126
C 349.959 899.166 308.949 1196.88 326.435 1352.92
C 343.921 1508.96 360.853 1694.26 464.524 1950.07
C 568.196 2205.88 713.904 2334.59 777.34 2401.28
C 840.776 2467.96 1002.66 2566.78 1084.52 2598.3
C 1084.52 2598.3 1215.37 2645.29 1336.74 2644.51
M 1318.43 1664.27
C 1318.43 1664.27 1464.11 1551.71 1425.52 1314.01
C 1386.92 1076.3 1275.12 1040.22 1259.24 1030.63
C 1259.24 1030.63 1140.06 1096.27 1139.47 1338.33
C 1139.47 1338.33 1143.44 1558.25 1318.43 1664.27
"/>
</g>
</svg>
```

**Reply by ciochu** [reply]
thank you @snibgo. You fully answered my question. ;) 

The only concern I have is how possible is that I can see Clipping path details for that [file](https://images.salsify.com/image/upload/s--_Jbe4hVk--/e8c7c13fd8b971386e811143c133abddad5632c4) but conversion can not be done to transparent PNG (i failed using various methods). 

 ` Clipping path:
<?xml version="1.0" encoding="iso-8859-1"?>
<svg xmlns="http://www.w3.org/2000/svg" width="2992" height="3000">
<g>
<path fill-rule="evenodd" style="fill:#000000;stroke:#000000;stroke-width:0;stroke-antialiasing:false" d="
M 1512.51 229.614
C 2173.87 243.32 2710 773.288 2710 1443.94
C 2699.34 2099.37 2173.87 2658.27 1512.51 2658.27
C 850.159 2665.27 327.685 2126.78 317.026 1443.94
C 317.026 773.288 822.752 215.046 1512.51 229.614
"/>
</g>
</svg>

  Profiles:
    Profile-8bim: 8880 bytes
    Profile-iptc: 15 bytes
      City[1,90]: 0x00000000: 254700                                        -%G
      unknown[2,0]:
    Profile-xmp: 14589 bytes
  Properties:
    8BIM:1999,2998:#1: <?xml version="1.0" encoding="iso-8859-1"?>
<svg xmlns="http://www.w3.org/2000/svg" width="2992" height="3000">
<g>
<path fill-rule="evenodd" style="fill:#000000;stroke:#000000;stroke-width:0;stroke-antialiasing:false" d="
M 1512.51 229.614
C 2173.87 243.32 2710 773.288 2710 1443.94
C 2699.34 2099.37 2173.87 2658.27 1512.51 2658.27
C 850.159 2665.27 327.685 2126.78 317.026 1443.94
C 317.026 773.288 822.752 215.046 1512.51 229.614
"/>
</g>
</svg>
`
